### ISSUE TITLE
Heap Snapshots crash when attempting to create

### ISSUE THREAD

**Issue by paulcbetts** [electron/electron#8791]
* Electron version: 1.6.1, on both macOS and Windows

The following code seems to crash any renderer process hard on apps I've tested on (Slack Desktop, unreleased project X):

```js
process.atomBinding('v8_util').takeHeapSnapshot();
```

https://github.com/enlight/v8-profiler seems to have a similar effect when trying to take a heap snapshot as well.

**Comment by kevinsawicki** [comment]
Here is the stack trace on macOS:

```
Thread 0 Crashed:: CrRendererMain  Dispatch queue: com.apple.main-thread
0   com.github.electron.framework 	0x000000010f0c7e98 atom::AtomBindings::Crash() + 8 (atom_bindings.cc:168)
1   com.github.electron.framework 	0x000000010f0c7e81 atom::(anonymous namespace)::FatalErrorCallback(char const*, char const*) + 209 (atom_bindings.cc:76)
2   libv8.dylib                   	0x0000000110d84ba7 v8::Object::SlowGetAlignedPointerFromInternalField(int) + 247
3   libblink_core.dylib           	0x000000011256f8f1 0x111a18000 + 11892977
4   libv8.dylib                   	0x000000011122e6dd 0x110d5e000 + 5048029
5   libv8.dylib                   	0x0000000111262c0a 0x110d5e000 + 5262346
6   libv8.dylib                   	0x000000011127221f 0x110d5e000 + 5325343
7   libv8.dylib                   	0x000000011131fa75 0x110d5e000 + 6036085
8   libv8.dylib                   	0x000000011131f1ea 0x110d5e000 + 6033898
9   libv8.dylib                   	0x0000000111324ee6 0x110d5e000 + 6057702
10  libv8.dylib                   	0x00000001113258c7 0x110d5e000 + 6060231
11  libv8.dylib                   	0x0000000111325682 0x110d5e000 + 6059650
12  libv8.dylib                   	0x000000011132a664 0x110d5e000 + 6080100
13  libv8.dylib                   	0x000000011131cbcc 0x110d5e000 + 6024140
14  com.github.electron.framework 	0x000000010f0b4e0d (anonymous namespace)::TakeHeapSnapshot(v8::Isolate*) + 45 (atom_api_v8_util.cc:95)
```

**Comment by kevinsawicki** [comment]
The V8 message is:

```
Fatal error in V8: v8::Object::GetAlignedPointerFromInternalField() Not a Smi
```

**Comment by kevinsawicki** [comment]
Looks like this may have been failing for awhile, @paulcbetts did you see this work in previous versions?

https://github.com/electron/electron/commit/7889e2750f3f6ab7ee3a9a6a9953827a8dca27e5

**Comment by kevinsawicki** [comment]
Looks like this may be fixed upstream via https://codereview.chromium.org/2669393002 referenced from https://bugs.chromium.org/p/chromium/issues/detail?id=661223

Building libcc now with that patch to confirm.

**Comment by jingman** [comment]
I believe I have the same issue (crash on heap profile), although I have a different set of error messages.

Since my app requires that I rebuild some native modules against a particular version of Electron, what's the best way for me to implement this patch myself without waiting for the next official release? Is there any way to tell if a release with this fix is imminent?

**Comment by kevinsawicki** [comment]
> Is there any way to tell if a release with this fix is imminent?

It should be released today in 1.6.4 beta today, there isn't an easy way for you to apply this patch locally.

**Comment by jingman** [comment]
Incredible! Thanks for the notice!!
